### ISSUE TITLE
Android Studio updated gradle version of my project and now flutter run fails

### ISSUE THREAD

**Issue by goderbauer** [flutter/flutter#9842]
I opened my Flutter project in Android Studio to edit some native Android code. Android Studio told me that I should immediately update the gradle version of my project because of important bug fixes (the messages sounded rather urgent). I foolishly clicked "upgrade gradle" and now my project doesn't build with `flutter run` anymore 😞 

```shell
C:\src\firebase_messaging\example [master]> flutter run
Running 'flutter packages get' in example...          0.7s
Launching lib/main.dart on Nexus 6P in debug mode...
FAILURE: Build failed with an exception.

* Where:
Build file 'C:\src\firebase_messaging\example\android\app\build.gradle' line: 14

* What went wrong:
A problem occurred evaluating project ':app'.
> Failed to apply plugin [id 'com.android.application']
   > Minimum supported Gradle version is 3.3. Current version is 3.2. If using the gradle wrapper, try editing the distributionUrl in C:\src\firebase_messaging\example\android\gradle\wrapper\gradle-wrapper.properties to gradle-3.3-all.zip

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.
Error running gradle: Exit code 1 from: C:\Program Files\Android\Android Studio\gradle\gradle-3.2\bin\gradle.bat app:properties
Running 'gradle assembleDebug'...

FAILURE: Build failed with an exception.

* Where:
Build file 'C:\src\firebase_messaging\example\android\app\build.gradle' line: 14

* What went wrong:
A problem occurred evaluating project ':app'.
> Failed to apply plugin [id 'com.android.application']
   > Minimum supported Gradle version is 3.3. Current version is 3.2. If using the gradle wrapper, try editing the distributionUrl in C:\src\firebase_messaging\example\android\gradle\wrapper\gradle-wrapper.properties to gradle-3.3-all.zip

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.
Gradle build failed: 1
```

Building and Running from Android Studio still works fine, but I miss that hot reload...

/cc @mit-mit @jakobr-google

**Comment by mit-mit** [comment]
While we are working on a real resolution to this problem, the following workaround seems to work:

Install gradle 3.3:
1. Download gradle-3.3 from https://services.gradle.org/distributions/gradle-3.2.1-bin.zip
1. Unzip file to the gradle installation dir
   * MacOS: `/Applications/Android Studio/Contents/gradle/`
   * Windows: `C:\Program Files\Android\Android Studio\gradle`

When opening in Android Studio, answer prompts like this:
* 'Gradle sync... Would you like the project to use gradle wrapper': Answer *Cancel*
* 'Gradle home: ..., Gradle location is unknown': Give path to gradle 3.3 from above
* 'Android gradle plugin update recommended': Answer 'Don't remind me again for this project'

**Comment by mravn-google** [comment]
Closed via #10928.
